tests/test_request_stream.py

```diff
@@ -336,6 +338,20 @@ async def post(request, id):
     assert response.status == 405
     assert "Method GET not allowed for URL /post/random_id" in response.text
 
+@pytest.mark.asyncio
+async def test_request_stream_unread(app):
+    """ensure no error is raised when leaving unread bytes in byte-buffer"""
+
+    err = None
+    protocol = HttpProtocol(loop=asyncio.get_event_loop(), app=app)
+    try:
+        protocol.request = None
+        protocol._body_chunks.append('this is a test')
+        await protocol.stream_append()
+    except AttributeError as e:
+        err = e
+
+    assert err is None and not protocol._body_chunks
```

Comment:
Please use `app.test_client` to test, instead of directly touching Sanic internals (like `_body_chunks`). Or did that not work for some reason (I know, streaming can be problematic with it).

Reply:
I wasn't sure how to access `steam_append` to do a proper test, so i just opted for accessing the object directly. I can try replacing references to body_chunks and instead just pass in a basic request-object into it using `on_headers_complete` to generate the request and then `on_body`. would that work?

Reply:
For this test, I am not sure that is necessary. At issue is the non-existence of the request object. If we want to add an additional end-to-end test, that works as well. But I think this is a valid solution.

Reply:
I suppose so. In this case I won't be taking this test to the streaming branch at all but it should suffice in master for now.

Reply:
> I wasn't sure how to access `steam_append` to do a proper test, so i just opted for accessing the object directly. I can try replacing references to body_chunks and instead just pass in a basic request-object into it using `on_headers_complete` to generate the request and then `on_body`. would that work?

You can get access through the `request` object.

```
request.transport.get_protocol()
```